pkg/cri/sbserver/container_create_linux.go

```diff
@@ -130,12 +130,14 @@ func (c *criService) containerSpecOpts(config *runtime.ContainerConfig, imageCon
 		specOpts = append(specOpts, oci.WithUser(userstr))
```

Comment:
Wondering why we're first going through the process above to convert username and group to a single string (`user:group` if both are present), only to then to parse everything again in `oci.WithUser()` to split user and group again and see if they're numeric values (should the "happy path" use `WithUIDGID()` or variants thereof?)

Reply:
I'm not sure why we did it like this. However, It looks as it is [from the beginning](https://github.com/containerd/containerd/commit/50c73e6dc550c2cdb579e303ac26394497f9f331). It should be refactored in another PR.

Reply:
Admitted, I haven't dug fully into all of the logic, but it looks like it's because there's a combination of a "strong typed" (`securityContext`) and fallback to the "weakly typed" (string) from the ` imageConfig.User`.

While trying to read the code, I found the logic to be complex / hard to read, which scared me a bit, as such cases can easily lead to bugs; too many conditions in a row make it hard to spot corner-cases, as well as the conversion from strong-typed to weak typed (followed by "trying to discover" if a `uid` or `username` was passed, or other combinations).

So, yes, perhaps out of scope for this PR (TBD?) but it would be good to clean this code up to make it more maintainable.

Reply:
nod.. I had to read it a few times take notes check test cases